src/elementary/src/split.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef NJOY_ELEMENTARY_SPLIT
+#define NJOY_ELEMENTARY_SPLIT
+
+// system includes
+#include <numeric>
+#include <string>
+
+// other includes
+
+namespace njoy {
+namespace elementary {
+
+  /**
+   *  @brief Split a string into a list of strings using a given separator
+   *
+   *  @param[in] string      the strings to be split
+   *  @param[in] separator   the separator to be used
+   */
+  std::vector< std::string > split( const std::string& string,
```

Comment:
We have a `split` function in [utility](https://github.com/njoy/utility), but I acknowledge that you don’t want to have any dependencies with this.